cmd/drone-docker/main.go

```diff
@@ -112,6 +112,12 @@ func main() {
 			Usage:  "don't start the docker daemon",
 			EnvVar: "PLUGIN_DAEMON_OFF",
 		},
+		cli.StringFlag{
+			Name:   "daemon.registry",
+			Usage:  "daemon registry",
+			Value:  "https://index.docker.io/v1/",
+			EnvVar: "DAEMON_REGISTRY,PLUGIN_REGISTRY,DOCKER_REGISTRY",
```

Comment:
For non-acr builds, `DAEMON_REGISTRY` will not be set and we will use `PLUGIN_REGISTRY`

Reply:
i am not particularly happy with this, as we are changing the default behaviour for the "docker.registry" why dont we just use the env variable PLUGIN_REGISTRY like we do for the other providers.

or is DAEMON_REGISTRY something totally new ?

Reply:
We have a hard assumption that `docker.registry` will point to a page where we can view the artifact. This is not the case for ACR. We get the url from Azure using the API implemented in the PR.

For kaniko, we have a separate struct for artifact, which is why it is straight forward there.

Ref:
1. https://github.com/drone/drone-kaniko/blob/main/kaniko.go#L239
2. https://github.com/drone/drone-kaniko/blob/main/cmd/kaniko-acr/main.go#L261

Reply:
We cannot update `PLUGIN_REGISTRY` as the push still needs to go to the input registry.

Reply:
I'm okay with adding `ArtifactRegistry` field to `docker.Daemon` and use it while writing the digest.

`drone.WritePluginArtifactFile(p.Daemon.RegistryType, p.ArtifactFile, p.Daemon.ArtifactRegistry, p.Build.Repo, digest, p.Build.Tags)`

Reply:
i was thinking of keeping the changes to main.go as https://github.com/drone-plugins/drone-docker/compare/acr...tphoney:drone-docker:acr?expand=1 and then adding logic into acr for what ever is needed. 
changing the handling inside of main.go and overriding what is standard is dangerous.
inside of the acr we can change as much as necessary.

Reply:
Moved to use `p.Daemon.ArtifactRegistry` which is a different path and not standard: https://github.com/drone-plugins/drone-docker/pull/395/commits/c64f19dec75948ec28095e4fc6e9ff5f12f2437e

Please approve and merge if it looks okay.